src/core/hle/kernel/thread.h

```diff
@@ -568,9 +570,62 @@ class Thread final : public SynchronizationObject {
         return has_exited;
     }
 
+    struct QueueEntry {
```

Comment:
```suggestion
    class QueueEntry {
```

Given this has private invariants.